File: 프로그래머스/Level2/카펫.py
# https://programmers.co.kr/learn/courses/30/lessons/42842

def solution(brown, yellow):
    def brown_number(a, b):
        return 2 * a + 2 * (b - 2)
    total = brown + yellow
    possible_set = set()
    for i in range(1, int(total ** (1/2)) + 1):
        if total % i == 0:
            possible_set.add(i)
    for a in possible_set:
        if brown == brown_number(a, total//a):
            if a >= total // a:
                return [a, total // a]
            return [total // a, a]

File: 프로그래머스/Level2/카펫.js
// https://programmers.co.kr/learn/courses/30/lessons/42842

function solution(brown, yellow) {
  const total = brown + yellow;

  for (let i = total; i >= 1; i--) {
    if (total % i !== 0) continue;
    if (2 * i + (2 * total) / i - 4 === brown) return [i, total / i];
  }
}
